src/tests/multihost/alltests/test_automount_from_bash.py

```diff
@@ -0,0 +1,995 @@
+            ldap_inst.add_entry(user_info, user_dn)
+            user_dn = f'cn=testkey{i},nisMapName=auto.share,ou=mount,{ds_suffix}'
+            user_info = {'objectClass': ['nisObject'.encode('utf-8'),
+                                         'top'.encode('utf-8')],
+                         'cn': f'testkey{i}'.encode('utf-8'),
+                         'nisMapName': 'auto.share'.encode('utf-8'),
+                         'nisMapEntry': f'-fstype=nfs,rw {master_server}:/export/foo/testkey{i}'.encode('utf-8')}
+            ldap_inst.add_entry(user_info, user_dn)
+        dumpmap_fixed_val = "automount - -dumpmaps | wc - l"
+        for i in range(1, 11):
+            dumpmap_current_val = "automount - -dumpmaps | wc - l"
+            assert dumpmap_fixed_val == dumpmap_current_val
+
+    @staticmethod
+    def test_autofs_segfaults(multihost, backupsssdconf):
+        """
+        :title: Automount segfaults in sss nss check header bz1123291
+        :id: 44af3d14-9897-11ed-a25c-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1123291
+        :setup:
+            1. The original SSSD configuration file is backed up and the new
+                configuration options are set using the sssd_conf method.
+                The options include the filter groups, filter users, and debug level for the nss section.
+                The script then sets additional parameters for the domain specified by ds_instance_name.
+            2. The SSSD cache is then cleared using the clear_sssd_cache method, and the autofs service is restarted.
+        :steps:
+            1. The script compiles and runs a C++ program called a1 100 times and not able to see the crash.
+        :expectedresults:
+            1. Should not crash
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        client.run_command(f'cp -f {SSSD_DEFAULT_CONF}.orig {SSSD_DEFAULT_CONF}')
+        tools.sssd_conf("nss", {'filter_groups': 'root',
+                                'filter_users': 'root',
+                                'debug_level': '9'}, action='update')
+        ldap_params = {'use_fully_qualified_names': False,
+                       'ldap_search_base': f'{ds_suffix}',
+                       'ldap_autofs_map_object_class': 'nisMap',
+                       'ldap_autofs_map_name': 'nisMapName',
+                       'ldap_autofs_entry_object_class': 'nisObject',
+                       'ldap_autofs_entry_key': 'cn',
+                       'ldap_autofs_entry_value': 'nisMapEntry'}
+        tools.sssd_conf(f'domain/{ds_instance_name}', ldap_params)
+        tools.clear_sssd_cache()
+        time.sleep(2)
+        client.run_command("service autofs restart")
+        client.run_command("g++ -lpthread -o a1 /tmp/a1.cpp")
+        client.run_command("chmod +x a1")
+        for _ in range(100):
+            client.run_command("./a1")
```

Comment:
Is any assertion expected in this testcase?

Reply:
No